### PR TITLE
feat(cli): add json test reporter

### PR DESCRIPTION
**Pull request by caspervonb**
This adds JSON as an output format for test results in the same style as the linter (non streaming).

Closes https://github.com/denoland/deno/issues/8550

### CODE REVIEW

**Comment by caspervonb** [comment]
Please merge #10428 first, as it changes the structures used.

File: cli/tests/test/deno_test_json.out [hunk 1]
```diff
@@ -0,0 +1,101 @@
+{
+  "events": [
```
Comment (nayeemrmn) [inline cli/tests/test/deno_test_json.out:2]: Discussed on discord. The output of `--json` shouldn't be events-based, it should print out a summary at the end. We should have a separate `--ndjson`/`--jsonl`/etc. flag which outputs test events as individual [line-delimited JSON](https://en.wikipedia.org/wiki/JSON_streaming#Line-delimited_JSON) objects. If anyone thinks we only need one of these, we should favour the latter. I think both are useful.

This pattern of having an overarching JSON object where nested array elements are chunked one-by-one isn't a real format and doesn't work well. Streaming consumers can't take advantage of the overarching JSON structure and will have a really annoying time parsing. Non-streaming consumers have no use for an events-based output and likely just need a summary. So this output doesn't serve anyone.

**Comment by caspervonb** [comment]
Punting this for a while because upcoming changes to the internal representation is going to affect the output.

**Comment by caspervonb** [comment]
Closing this for now, will revisit later.

**Pull request closed without merging**